job_creator.php

```diff
@@ -0,0 +1,270 @@
+                        $matrix['include'][] = $this->createJob(0, [
+                            'composer_args' => '--prefer-lowest',
+                            'phpunit' => true,
+                            'phpunit_suite' => $ts->getAttribute('name'),
+                        ]);
+                        $matrix['include'][] = $this->createJob(1, [
+                            'db' => 'pgsql',
+                            'phpunit' => true,
+                            'phpunit_suite' => $ts->getAttribute('name')
+                        ]);
+                        $matrix['include'][] = $this->createJob(3, [
+                            'db' => DB_MYSQL_80,
+                            'phpunit' => true,
+                            'phpunit_suite' => $ts->getAttribute('name')
+                        ]);
+                    }
+                }
+                // phpunit.xml has no defined testsuites
+                if (count($matrix['include']) == 0) {
+                    if ($simpleMatrix) {
+                        $matrix['include'][] = $this->createJob(0, [
+                            'phpunit' => true,
+                            'phpunit_suite' => 'all'
+                        ]);
+                    } else {
+                        $matrix['include'][] = $this->createJob(0, [
+                            'composer_args' => '--prefer-lowest',
+                            'phpunit' => true,
+                            'phpunit_suite' => 'all'
+                        ]);
+                        $matrix['include'][] = $this->createJob(1, [
+                            'db' => DB_PGSQL,
+                            'phpunit' => true,
+                            'phpunit_suite' => 'all'
+                        ]);
+                        $matrix['include'][] = $this->createJob(3, [
+                            'db' => DB_MYSQL_80,
+                            'phpunit' => true,
+                            'phpunit_suite' => 'all'
+                        ]);
+                    }
+                }
+            }
+            // skip phpcs on silverstripe-installer which include sample file for use in projects
+            if ((file_exists('phpcs.xml') || file_exists('phpcs.xml.dist')) && !preg_match('#/silverstripe-installer$#', $githubRepository)) {
+                $matrix['include'][] = $this->createJob(0, [
+                    'phplinting' => true
+                ]);
+            }
+            // phpcoverage also runs unit tests
+            // always run on silverstripe account
+            if ($run['phpcoverage'] || preg_match('#^silverstripe/#', $githubRepository)) {
+                if ($simpleMatrix) {
+                    $matrix['include'][] = $this->createJob(0, [
+                        'phpcoverage' => true
+                    ]);
+                } else {
+                    $matrix['include'][] = $this->createJob(2, [
+                        'db' => DB_MYSQL_57_PDO,
```

Comment:
Disagree, I think it's much cleaner to a separate block for each input

Reply:
In that case we should include a $phpIndex=2, db=DB_MYSQL_57_PDO job in the phpunit chunk, because those should be tested regardless of whether code coverage is included.

```php
$jobs[] = $this->createJob(2, [
    'db' => DB_MYSQL_57_PDO,
    'phpunit' => true,
    'phpunit_suite' => $suite,
]);
```